### DISCUSSION TITLE
Seeking CS 1 content

### DISCUSSION THREAD

**Discussion by dowobeha** [PrairieLearn/PrairieLearn#10586] [Course content sharing]
Hi, I'm teaching CS 201 (1st semester programming in modern C++) at the University of Alaska Fairbanks. Would anyone happen to have any content for a similar class that they'd be willing to share?

Topics:

- Intro to C++
- Basic types
  - int
  - double
  - char
  - bool
  - std::string
  - std::vector
- Control flow
  - if, then, else, else if
  - for-each loop
  - traditional C++ for loop
  - while loop
- Basic I/O
  - cout
  - cin
  - cerr
  - ostream operator overloading
  - istream operator overloading
- User-defined types
  - enum class
  - class
    - constructors
    - member variables
    - get and set methods

**Comment by ashallue** [comment]
I have content for a CS 1 course in Python.  This semester I am trying to write content for C++, and I'm willing to share when I do.  So far, I'm stuck trying to get the check suite to work.

--Andrew Shallue, Illinois Wesleyan University

**Reply by dowobeha** [reply]
Please feel free to get in touch with me. 

--Lane Schwartz, University of Alaska Fairbanks
firstname [dot] lastName @ alaska [dot] edu
